apps/api/src/services/data-provider/yahoo-finance/yahoo-finance.service.ts

```diff
@@ -253,21 +253,26 @@ export class YahooFinanceService implements DataProviderInterface {
     ) {
       if (isCurrency(aSymbol.substring(0, aSymbol.length - 3))) {
         return `${aSymbol}=X`;
-      } else if (
-        this.cryptocurrencyService.isCrypto(aSymbol) ||
-        this.cryptocurrencyService.isCrypto(aSymbol.replace('1', ''))
-      ) {
+      } else if (this.isCryptoCurrency(aSymbol)) {
         // Add a dash before the last three characters
         // BTCUSD  -> BTC-USD
         // DOGEUSD -> DOGE-USD
         // SOL1USD -> SOL1-USD
-        return aSymbol.replace('USD', '-USD');
+        return aSymbol.replace(new RegExp('-?(USD|EUR)$'), '-$1');
       }
     }
 
     return aSymbol;
   }
 
+  private isCryptoCurrency(aSymbol: string) {
```

Comment:
What do you want to improve exactly?

The current implementation is hacky, but now we would have to deal with `isCrypto()` and `isCryptoCurrency()` and two regular expressions.

Reply:
There are two usages of this method. See line 202.
Thie idea is to not have cryptocurrencies in the search result which you don't consider as a cryptocurrency.
If you don't like it, it might be an option to do the check with the `1` replacement in the `cryptocurrencyService`